### PR TITLE
Grouping methods in ExprTycker

### PR DESCRIPTION
**Pull request by HoshinoTented**
This PR groups methods in ExprTycker.

### CODE REVIEW

File: base/src/main/java/org/aya/tyck/ExprTycker.java [hunk 1]
```diff
@@ -48,6 +50,116 @@
...
+      if (type instanceof PiTerm pi && !pi.param().explicit() && needImplicitParamIns(e)) {
+        var implicitParam = new Term.Param(new LocalVar(Constants.ANONYMOUS_PREFIX), pi.param().type(), false);
+        var body = ctx.with(implicitParam, () -> inherit(e, pi.substBody(implicitParam.toTerm()))).wellTyped();
+        return new Result.Default(new LamTerm(LamTerm.param(implicitParam), body), pi);
+      } else return doInherit(e, type);
+    });
+  }
+
+  public @NotNull Result check(@NotNull Expr expr, @NotNull Term type) {
+    return inherit(expr, type);
```
[the remark this answers is not in the document]
Comment (ice1000) [reply base/src/main/java/org/aya/tyck/ExprTycker.java:160]: Yes, please!

**Comment by ice1000** [comment]
Thank you! 🙏

File: base/src/main/java/org/aya/tyck/pat/ClauseTycker.java [hunk 2]
```diff
@@ -169,13 +169,12 @@ private static Pat.Preclause<Term> checkRhs(@NotNull ExprTycker exprTycker, @Not
...
-          // as we bind local variables in the pattern checker,
-          // and in case the patterns are malformed, some bindings may
-          // not be added to the localCtx of tycker, causing assertion errors
-          ? new ErrorTerm(e, false)
-          : exprTycker.check(e, lhsResult.type).wellTyped());
+        return lhsResult.preclause.expr().map(e -> {// In case the patterns are malformed, do not check the body
+// as we bind local variables in the pattern checker,
+// and in case the patterns are malformed, some bindings may
+// not be added to the localCtx of tycker, causing assertion errors
+          return lhsResult.hasError ? new ErrorTerm(e, false) : exprTycker.inherit(e, lhsResult.type).wellTyped();
```
Comment (imkiva) [inline base/src/main/java/org/aya/tyck/pat/ClauseTycker.java:176]: Code formatting seems broken

Comment (HoshinoTented) [reply base/src/main/java/org/aya/tyck/pat/ClauseTycker.java:176]: 😱

**Pull request merged**